tests/lib/rules/forbid-foreign-prop-types.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Tests for forbid-foreign-prop-types
+ */
+'use strict';
+
+// -----------------------------------------------------------------------------
+// Requirements
+// -----------------------------------------------------------------------------
+
+var rule = require('../../../lib/rules/forbid-foreign-prop-types');
+var RuleTester = require('eslint').RuleTester;
+
+var parserOptions = {
+  ecmaVersion: 6,
+  ecmaFeatures: {
+    experimentalObjectRestSpread: true,
+    jsx: true
+  }
+};
+
+require('babel-eslint');
+
+// -----------------------------------------------------------------------------
+// Tests
+// -----------------------------------------------------------------------------
+
+var ERROR_MESSAGE = 'Using another component\'s propTypes is forbidden';
+
+var ruleTester = new RuleTester();
+ruleTester.run('forbid-foreign-prop-types', rule, {
+
+  valid: [{
+    code: 'import { propTypes } from "SomeComponent";',
+    parser: 'babel-eslint'
+  }, {
+    code: 'import { propTypes as someComponentPropTypes } from "SomeComponent";',
+    parser: 'babel-eslint'
+  }],
```

Comment:
It might be worth adding some valid cases like:

```js
const foo = propTypes
foo(propTypes)
foo + propTypes
const foo = [propTypes]
const foo = { propTypes }
```

And maybe even junk like this (note the lowercase):

```js
foo.propTypes
foo['propTypes']
const { propTypes } = foo
```